firefox_ui_harness/testcases/update.py

```diff
@@ -0,0 +1,133 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from ..testcase import FirefoxTestCase
+
+
+class UpdateTestCase(FirefoxTestCase):
+
+    def __init__(self, *args, **kwargs):
+        FirefoxTestCase.__init__(self, *args, **kwargs)
+
+        self.target_buildid = kwargs.pop('update_target_buildid')
+        self.target_version = kwargs.pop('update_target_version')
+
+        self.update_channel = kwargs.pop('update_channel')
+        self.update_mar_channels = set(kwargs.pop('update_mar_channels'))
+
+        self.updates = []
+
+    def setUp(self, is_fallback=False):
+        FirefoxTestCase.setUp(self)
+
+        self.current_update_index = 0
```

Comment:
This exists on purpose. I told multiple times that we want to support updates in multiple steps with Marionette. We haven't finished this off in Mozmill, so this is a straight port-over, which I or someone else will pick up once this PR has been merged. So lets leave this in to not have to rewrite all the logic twice.